packages/gateway/src/commands/supergraph.ts

```diff
@@ -249,7 +250,9 @@ export const addCommand: AddCommand = (ctx, cli) =>
         process.exit(1);
       }
       return runSupergraph(ctx, config);
-    });
+    })
+    .allowUnknownOption(process.env.NODE_ENV === 'test')
```

Comment:
Can you please elaborate on this change?

Reply:
When running tests, `NODE_ENV` passed as `test`, and this allows us to accept extra CLI arguments.
We need this but why?
Because during tests random ports are used and the CLI needs to know which ports are used for which services.
This was missing for GW CLI but only available for Mesh Compose. So in order to have a test self hosting service, we need to pass our test environment's port to the GW CLI.